[plugin/rewrite] Refactor to satisfy security scan

yongtang opened the pull request:
Re-arrange the logic so that it will not hit the false positive of DAST scan result from LTGM.com:
https://lgtm.com/projects/g/coredns/coredns/?mode=list

The re-arrange also avoid an extra assign (not a big diff, though).

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Fix an array out of bounds index issue

This PR tries to address an array out-of-bounds index issue
that may pose security risks, by adding array length check.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

chrisohaver reviewed (changes requested):
I don't think the range check is currently necessary due to the logic above it.

yongtang committed:
Re-arrange the logic to prevent false positive of security scan.

While the boundary check is not needed, it does cause a false positive
with security scan of LGTM. This PR re-arrange the logic to avoid the
false positive.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

chrisohaver reviewed (approved):
LGTM - I'm not offended by nested ifs and else statements.
